core/backend/src/assets/Settings/Schemas/Gcs.schema.json

```diff
@@ -1,32 +1,28 @@
 {
-  "$id": "http://itwinjs.org/json-schema/gcs",
-  "$schema": "http://json-schema.org/draft-07/schema",
```

Comment:
Using the JSON spec was nice because we were [generating types from our settings schemas](https://github.com/iTwin/studio/blob/main/packages/framework/src/common/settings/SettingsSchema.d.ts) automatically. It was useful, for example, if my setting was an enum, while developing my app I'll see the enum values in my editor. Likewise if my setting is an object, I'll know the keys.

Now with this custom schema (with settingDefs and typeDefs), [we won't be able to do that automatically](https://github.com/iTwin/studio/blob/main/packages/framework/scripts/create-settings-schema-type.js).

Reply:
we should talk. I don't know if you can do what i did with the "extends" keyword in JSONSchema. Maybe you can, that whole JSONSchema thing seems like it was invented by amateurs haphazardly, so I gave up.

Reply:
Keith, while I did find a way to make it work with JSON Schema, its probably not worth the trade-off. Yes, you'll be able to use JSON-to-Typescript and have the types, but now you have to deal with JSON Schema and all the extra baggage and properties that we don't use or need.